src/hooks/useSingleStageHandler.ts

```diff
@@ -76,25 +77,56 @@ export const useSingleStageHandler = () => {
   useEffect(() => {
     const fetchStage = async () => {
       if (parseInt(stageAddress, 16) != 0) {
-        const stageContract = await getSmartContractWithProvider(
-          stageAddress,
-          provider,
-          JSON.stringify(StageContract.abi)
-        );
-        const stageIpfsKey = await stageContract.ipfsKey();
-        const expiryBlock = await stageContract.expiryBlock();
-        const startBlock = await stageContract.startBlock();
-        const isComplete = await stageContract.isComplete();
-        const projectOwner = await stageContract.projectOwner();
-        const stageIpfsData = await getStageData(stageIpfsKey);
-        const isSuccess = await stageContract.isSuccess();
-        const yays = await stageContract.yays();
-        const totalVotes = await stageContract.totalVotes();
-        const totalCommitted = await stageContract.totalCommitted();
-        const votingPeriod = await stageContract.votingPeriod();
-        const expiryBlockInt = parseInt(expiryBlock.toString());
-        const votingPeriodInt = parseInt(votingPeriod.toString());
-        const voted = await stageContract.voted(address);
+        let stageIpfsKey;
+        let expiryBlock;
+        let startBlock;
+        let isComplete;
+        let projectOwner;
+        let stageIpfsData;
+        let isSuccess;
+        let yays;
+        let totalVotes;
+        let totalCommitted;
+        let votingPeriod;
+        let expiryBlockInt;
+        let votingPeriodInt;
+        let voted;
+
+        if (state) {
+          isComplete = state.stageContract.isComplete;
+          isSuccess = state.stageContract.isSuccess;
+          startBlock = state.stageContract.startBlock;
+          expiryBlock = state.stageContract.expiryBlock;
+          yays = state.stageContract.yays;
+          totalVotes = state.stageContract.totalVotes;
+          totalCommitted = state.stageContract.totalCommitted;
+          projectOwner = state.stageContract.projectOwner;
+          votingPeriod = state.stageContract.votingPeriod;
+          stageIpfsData = state.stage;
+          expiryBlockInt = parseInt(expiryBlock.toString());
+          votingPeriodInt = parseInt(votingPeriod.toString());
+          voted = false;
+        } else {
+          const stageContract = await getSmartContractWithProvider(
```

Comment:
Can you do this?
if (state) {
  stageContract = state.stageContract
} else {
  stageContract = await getSmartContractWithProvider()
}
?

Reply:
Done